src/main/java/ch/wisv/areafiftylan/users/model/User.java

```diff
@@ -35,67 +33,48 @@
 
 
 @Entity
-@Table(uniqueConstraints = { @UniqueConstraint(name = "username", columnNames = { "username" }) })
-@EqualsAndHashCode
+@Data
+@RequiredArgsConstructor
 @NoArgsConstructor
+@Table(uniqueConstraints = { @UniqueConstraint(name = "username", columnNames = { "username" }) })
 public class User implements Serializable, UserDetails {
 
-    @Column(nullable = false)
-    private String passwordHash;
-
+    @NonNull
     @Column(nullable = false)
     @JsonView(View.NoProfile.class)
     @Email(message = "Username should be a valid Email!")
-    @Getter
-    @Setter
     private String username;
 
+    @NonNull
+    @Column(nullable = false)
+    private String passwordHash;
+
     @OneToOne(targetEntity = Profile.class, cascade = CascadeType.ALL)
     @JsonView(View.Public.class)
-    @Getter
-    private Profile profile;
+    private Profile profile = new Profile();
 
     @GeneratedValue
     @Id
-    @Getter
     private Long id;
 
     @JsonIgnore
     @ElementCollection(targetClass = Role.class, fetch = FetchType.EAGER)
     @Enumerated(EnumType.STRING)
     @CollectionTable(name = "user_role")
-    private Set<Role> roles;
+    private Set<Role> roles = new HashSet<>(Sets.newHashSet(Role.ROLE_USER));
```

Comment:
While we're changing this, I'm not sure this is the best place to assign default Roles. Perhaps move this to the Service?